script.js

```diff
@@ -40,11 +48,15 @@ class ToDoList {
         this.saveToLocalStorage();
     }
     saveToLocalStorage() {
-        window.localStorage.setItem("to-do-list", JSON.stringify(this.toDos))
+        // Check if useLocalStorage is enabled for this instance, so we only save to
+        // local storage if the list requests it.
+        if (this.useLocalStorage) {
+            window.localStorage.setItem(this.listName, JSON.stringify(this.toDos))
```

Comment:
Same for listName, need also to assign it to the constructor
` this.listName = listName;`